common/persistence/sql/sqlplugin/history_transfer_tasks.go

```diff
@@ -0,0 +1,63 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package sqlplugin
+
+import "database/sql"
+
+type (
+	// TransferTasksRow represents a row in transfer_tasks table
+	TransferTasksRow struct {
+		ShardID      int32
+		TaskID       int64
+		Data         []byte
+		DataEncoding string
+	}
+
+	// TransferTasksFilter contains the column names within transfer_tasks table that
+	// can be used to filter results through a WHERE clause
+	TransferTasksFilter struct {
+		ShardID int32
+		TaskID  int64
+	}
+
+	// TransferTasksRangeFilter contains the column names within transfer_tasks table that
+	// can be used to filter results through a WHERE clause
+	TransferTasksRangeFilter struct {
+		ShardID   int32
+		MinTaskID int64
+		MaxTaskID int64
+	}
+
+	// HistoryTransferTask is the SQL persistence interface for history nodes and history transfer tasks
+	HistoryTransferTask interface {
+		InsertIntoTransferTasks(rows []TransferTasksRow) (sql.Result, error)
+		// SelectFromTransferTasks returns rows that match filter criteria from transfer_tasks table.
+		SelectFromTransferTasks(filter TransferTasksFilter) ([]TransferTasksRow, error)
```

Comment:
Is this now a point lookup for which both `SharID` and `TaskID` have to be provided? If that's the case, do we even need `TransferTasksFilter` anymore? Should `SelectFromTransferTasks()` take `SharID` and `TaskID` as two arguments instead?

Reply:
we can certainly do that.
from my understanding, the filter thing (single parameter vs multiple parameters) is to reduce the amount of work when refactoring / adding variable to the filter.

currently the convention of the SQL side code logic is using a single parameter

Reply:
If that's the convention we want to follow, that's fine with me. Especially if this is not exposed in the SDK.

Reply:
currently the interfaces are purely internal
more need to be done to make the SQL / NoSQL persistence interface public (modularized)